product_input_interface/input_interface.py

```diff
@@ -0,0 +1,82 @@
+from product_input_interface.allowed_products import WarehouseProducts
+
+
+class InputInterface:
+
+    def __init__(self):
+        self.WAREHOUSE = WarehouseProducts()
+        self.QUIT_LOOP = "quit"
+
+    @staticmethod
+    def __user_input_product_name():
+        return input("Please enter product name: ")
+
+    @staticmethod
+    def __user_quantity_of_product():
+        return input("Please enter quantity: ")
+
+    @staticmethod
+    def __user_input_product_type():
+        return input("Please enter product type:")
+
+    def __get_product_type(self):
+        while True:
+            self.WAREHOUSE.print_allowed_product_types()
+            product_type = self.__user_input_product_type()
+            if product_type.lower() == self.QUIT_LOOP:
+                return self.QUIT_LOOP
+
+            product_type_is_valid = self.WAREHOUSE.check_product_type(product_type.lower())
+            if product_type_is_valid:
+                return product_type
+
+    def __get_product_name(self):
+        while True:
+            product_name = self.__user_input_product_name()
+            if product_name.lower() == self.QUIT_LOOP:
+                return self.QUIT_LOOP
+
+            product_name_is_valid = self.WAREHOUSE.check_product_name_length(product_name)
+            if product_name_is_valid:
+                return product_name
+
+    def __get_product_quantity(self):
+        while True:
+            quantity = self.__user_quantity_of_product()
+            if quantity == self.QUIT_LOOP:
+                return self.QUIT_LOOP
+
+            try:
+                if int(quantity) == 0:
+                    print("You cant add 0 quantity")
+                else:
+                    return int(quantity)
+            except ValueError as e:
+                print(f"You need to input natural number. {e}")
```

Comment:
that code won't throw error if I will provide value like e.g. `2.3` but it will silently assume `2` in such case, dropping the float. I would change it a little to tell the user that we can't do such thing